mongo/countopt/etimatedcountopt.go

```diff
@@ -0,0 +1,60 @@
+package countopt
+
+import (
+	"github.com/mongodb/mongo-go-driver/core/option"
+)
+
+var estimatedDocumentCountBundle = new(EstimatedDocumentCountBundle)
+
+// EstimatedDocumentCount is options for the estimatedDocuemntCount() function
+type EstimatedDocumentCount interface {
+	estimatedCount()
+	ConvertEstimateDocumentCountOption() option.EstimatedDocumentCountOptioner
+}
+
+// EstimatedDocumentCountBundle is a bundle of EstimatedDocumentCount options
+type EstimatedDocumentCountBundle struct {
+	option EstimatedDocumentCount
+	next   *EstimatedDocumentCountBundle
+}
+
+// Implement the EstimateDocumentCount interface
+func (cb *EstimatedDocumentCountBundle) estimatedCount() {}
+
+// ConvertEstimateDocumentCountOption implements the EstimatedDocumentCount interface
+func (cb *EstimatedDocumentCountBundle) ConvertEstimateDocumentCountOption() option.EstimatedDocumentCountOptioner {
+	return nil
+}
+
+// BundleEstimatedDocumentCount bundles EstimatedDocumentCount options
+func BundleEstimatedDocumentCount(opts ...EstimatedDocumentCount) *EstimatedDocumentCountBundle {
+	head := estimatedDocumentCountBundle
+
+	for _, opt := range opts {
+		newBundle := EstimatedDocumentCountBundle{
+			option: opt,
+			next:   head,
+		}
+
+		head = &newBundle
+	}
+
+	return head
+}
+
+// MaxTimeMs adds an option to specify the maximum amount of time to allow the operation to run.
+func (cb *EstimatedDocumentCountBundle) MaxTimeMs(i int32) *EstimatedDocumentCountBundle {
+	bundle := &EstimatedDocumentCountBundle{
+		option: MaxTimeMs(i),
+		next:   cb,
+	}
+
+	return bundle
+}
+
+// ConvertEstimateDocumentCountOption implements the Count interface.
+func (opt OptMaxTimeMs) ConvertEstimateDocumentCountOption() option.EstimatedDocumentCountOptioner {
```

Comment:
These methods should be in countopt.go with the declaration of OptMaxTimeMs.

Reply:
Done.